modules/DragonridingVigor.lua

```diff
@@ -30,20 +130,42 @@ function DragonridingVigor.prototype:init()
 	self.bTreatEmptyAsFull = false
 	self.runeWidth = self.runeHeight
 	self.shouldRegisterDisplayPower = false
+
+	self.mode = "widget" -- "widget" (old) or "charges" (new)
+	self.chargeSpellID = nil
 end
 
 function DragonridingVigor.prototype:Enable(core)
-	self.numRunes = UnitPowerMax(self.unit, unitPowerType)
-	self.runeCoords = { }
-	for i = 1, self.numRunes do
-		self:SetupNewRune(i)
+	-- Decide mode:
+	-- In 11.2.7+ Blizzard removed Vigor and moved Skyriding to shared charges.
+	-- Even if the old Vigor widget still exists, it may stay hidden (shownState=0),
+	-- so we prefer charge mode on modern builds.
+	local preferCharges = (BUILD_NUMBER and BUILD_NUMBER >= FORCE_CHARGES_BUILD) or (not self:UsingOldVigorWidgets())
+
+	if preferCharges then
+		self.mode = "charges"
+		self.chargeSpellID = self:GetChargeSpell()
+		local _, maxCharges = self.chargeSpellID and SafeGetSpellCharges(self.chargeSpellID) or nil
+		self:EnsureRuneCount(maxCharges or 6)
+	else
+		self.mode = "widget"
+		self.numRunes = UnitPowerMax(self.unit, unitPowerType) or 0
+		self:EnsureRuneCount(self.numRunes)
 	end
 
 	DragonridingVigor.super.prototype.Enable(self, core)
 	self:Show(false)
 
+	-- Widget-driven show/hide + recharge (legacy)
 	self:RegisterEvent("UNIT_AURA", "CheckShouldShow")
 	self:RegisterEvent("UPDATE_UI_WIDGET", "UpdateVigorRecharge")
+
+	-- Charge/power-driven updates (modern)
+	self:RegisterEvent("PLAYER_MOUNT_DISPLAY_CHANGED", "CheckShouldShow")
+	self:RegisterEvent("SPELL_UPDATE_CHARGES", "UpdateVigorRecharge")
+	self:RegisterEvent("SPELL_UPDATE_COOLDOWN", "UpdateVigorRecharge")
+	self:RegisterEvent("UNIT_POWER_UPDATE", "UpdateRunePower")
+	self:RegisterEvent("UNIT_POWER_FREQUENT", "UpdateRunePower")
```

Comment:
Are we sure all these events are available in any client version that may try to run this code? Attempting to register an event that doesn't exist will throw an error, so if we need to gate this registration on something, we should.